Sources/Paywalls/Components/Transitions/PaywallTransition.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallTransition.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/20/25.
+// swiftlint:disable missing_docs
+
+import Foundation
+
+public extension PaywallComponent {
+    struct Transition: PaywallComponentBase {
+        public let type: TransitionType
+        public let displacementStrategy: DisplacementStrategy
+        public let animation: PaywallComponent.Animation?
+
+        init (
+            type: TransitionType = .fade,
+            displacementStrategy: DisplacementStrategy = .greedy,
+            animation: PaywallComponent.Animation? = nil
+        ) {
+            self.type = type
+            self.displacementStrategy = displacementStrategy
+            self.animation = animation
+        }
+
+        public init(from decoder: any Decoder) throws {
+            let passthrough = try TransitionCodingContainer(from: decoder)
+
+            self.type = TransitionType.from(passthrough.type) ?? .fade
+            self.displacementStrategy = passthrough.displacementStrategy
+            self.animation = passthrough.animation
+        }
+
+        public func encode(to encoder: any Encoder) throws {
+            try TransitionCodingContainer(
+                type: type.codingContainer,
+                displacementStrategy: displacementStrategy,
+                animation: animation
+            )
+            .encode(to: encoder)
+        }
+    }
+
+    internal struct TransitionCodingContainer: Codable {
+        let type: TransitionTypeContainer
+        let displacementStrategy: DisplacementStrategy
+        let animation: PaywallComponent.Animation?
+    }
```

Comment:
I believe in other `PaywallComponent`s we have `CodingKeys` instead of a `CodingContainer` 

```swift
private enum CodingKeys: String, CodingKey {
    case type
    case displacementStrategy
    case animation
}
```

I think it's equivalent but double checking if it's worth it to use `CodingKeys` for consistency